code-review.js

```diff
@@ -1,60 +1,119 @@
-	this.Pop = function() {
-		maxKey = Math.max(Object.keys(this.store));
+	/**
+	 * Returns the oldest-added value with the highest priority
+	 * @returns {*} - value removed
+	 */
+	this.pop = function() {
+		if (this.count === 0) {
+			return false;
+		}
+		var maxKey = Math.max.apply(null, this.getAllPriorities());
+		var poppedValue = this.queue[maxKey].shift();
 		this.count--;
-		return this.store[maxKey].shift();
+		this.removeEmptyPriority(maxKey);
+
+		return poppedValue
 	};
 
+	/**
+	 * Returns the number of elements in queue
+	 * @returns {Number} 
+	 */
 	this.length = function() {
 		return this.count;
-	}
-}
+	};
+};
 
-PriorityQueue.prototype.get_all_priorities = function() {
-	return Object.keys(this.store);
-}
+/**
+ * Returns all the priorities in descending order
+ * @returns {Array}
+ */
+PriorityQueue.prototype.getAllPriorities = function() {
+	return Object.keys(this.queue).sort(function(a, b){
+		return b - a;
+	});
+};
 
-// iterates through all the queue elements in priority-then-FIFO order
-PriorityQueue.prototype.forEach = function(callback) {
-	var keys = Object.keys(this.store).sort();
+/**
+ * Iterates through all the queue elements in priority-then-FIFO order
+ * 
+ * @param {function} callback - Callback function to be used on every element
+ */
+PriorityQueue.prototype.priorityForEach = function(callback) {
+	var allPriorities = this.getAllPriorities();
+	allPriorities.forEach(function(priority) {
+		for (var valueIndex = 0; valueIndex < this.queue[priority].length; valueIndex++) {
+			callback(this.queue[priority][valueIndex]);
+		}
+	}, this);
+};
 
-	for (var a = keys.length; a > 0; a--) {
-		for (var b = 0; b < this.store[a].length; b++)
-			callback(this.store[a][b]);
+/**
+ * Tests if a priority's array is empty and deletes the priority if not
+ * 
+ * @param {Number} priority - Priority to be tested
+ */
+PriorityQueue.prototype.removeEmptyPriority = function(priority) {
+	if (this.queue[priority].length === 0){
+		delete this.queue[priority];
 	}
-}
+};
 
+/**
+ * Changes the Priority to a different key value.
+ *
+ * @param {*} value - The Item we wish to change
+ * @param {Number} newPriority - Positive integer
+ * @returns {Boolean} - true if value exists, else returns false
+ */
 PriorityQueue.prototype.changePriority = function(value, newPriority) {
 	var foundItem = false;
+	var allPriorities = this.getAllPriorities();
 
-	this.store.forEach(function(bucket) {
-		bucket.forEach(function(item, index) {
-			if (item === value) {
-				bucket.splice(index, 1);  // remove the item
+	allPriorities.some(function(priority) {
+		for (var valueIndex = 0; valueIndex < this.queue[priority].length; valueIndex++) {
+			if (this.queue[priority][valueIndex] === value) {
+				this.queue[priority].splice(valueIndex, 1);  // remove the item
 				this.add(value, newPriority);
+				this.count--;
 				foundItem = true;
-				return false;  // early exit from forEach

```

Comment:
Impossible to "early exit" `forEach()`, the code in the loop won't be interpreted after the `return false`, but the loop still iterates. 

The alternative is using `some()`. :stuck_out_tongue: nice gotcha.